tsconfig.json

```diff
@@ -1,26 +1,21 @@
 {
     "compilerOptions": {
-        "outDir": "dist",
-        "module": "commonjs",
+        "module": "preserve",
         "target": "ES2022",
-        "lib": ["ES2022", "dom"],
-        "sourceMap": true,
-        "declaration": true,
-        "declarationMap": false,
+        "lib": ["ES2022"],
         "types": ["node", "vitest/globals"],
-        "strict": true,
+        "moduleDetection": "force",
+        "verbatimModuleSyntax": true,
         "skipLibCheck": true,
-        "moduleResolution": "node",
+        "esModuleInterop": true,
+        "resolveJsonModule": true,
+        "noEmit": true,
```

Comment:
why do we use noEmit by default?

Reply:
typescript is used only for typechecking now, building is handled by tsup